vertx-web/src/main/java/io/vertx/ext/web/impl/Utils.java

```diff
@@ -331,13 +335,45 @@ public static Buffer readResourceToBuffer(String resource) {
     }
   }
 
+  private static final Charset DEFAULT_CHARSET = Charset.defaultCharset();
+
   /*
-  Reads from file or classpath
+  Reads from file or classpath using the default charset, replacing CRLF line endings with LF ones
    */
   public static String readFileToString(Vertx vertx, String resource) {
+    return readFileToString(vertx, resource, DEFAULT_CHARSET);
+  }
+
+  private static final boolean convertLineEndings = System.lineSeparator().equals("\r\n");
+
+  /*
+  Reads from file or classpath using the given charset, replacing CRLF line endings with LF ones
+   */
+  public static String readFileToString(Vertx vertx, String resource, Charset charset) {
     try {
       Buffer buff = vertx.fileSystem().readFileBlocking(resource);
-      return buff.toString();
+      if (!convertLineEndings) {
+        return buff.toString(charset);
+      }
+      int buffLen = buff.length();
+      ByteBuf byteBuf = buff.getByteBuf();
```

Comment:
What will be the advantage of using the underlying netty API here, IMO we should keep with the vert.x API since it would address eventual changes underlying and later we're just iterating all bytes so one could achieve the same with a loop on the vert.x buffer.